dartagnan/src/test/java/com/dat3m/dartagnan/spirv/header/SpirvHeaderSingleTest.java

```diff
@@ -0,0 +1,132 @@
+package com.dat3m.dartagnan.spirv.header;
+
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.parsers.program.ParserSpirv;
+import com.dat3m.dartagnan.program.Program;
+import com.dat3m.dartagnan.program.memory.Location;
+import com.dat3m.dartagnan.program.specification.AssertBasic;
+import com.dat3m.dartagnan.program.specification.AssertCompositeAnd;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class SpirvHeaderSingleTest {
+    public SpirvHeaderSingleTest() {
```

Comment:
And how about a few test cases for the precedence of and/or and brackets

Reply:
Added